### PR TITLE
openal: add 1.21.0

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **openal/1.21.0**

Overrides #3644

- [x] FIXME: make example pure C
--- 

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/openal/all/conanfile.py [hunk 1]
```diff
@@ -31,12 +31,32 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+    @property
+    def _minimum_compiler_supports_cxx14(self):
+        return  {
+            "Visual Studio": "15",
+        }.get(str(self.settings.compiler))
```
Comment (prince-chrismc) [inline recipes/openal/all/conanfile.py:38]: should this have more compilers?

Comment (madebr) [reply recipes/openal/all/conanfile.py:38]: Yes, going to work on it now.
I was on Windows and could not test the Linux compilers.

File: recipes/openal/all/conanfile.py [hunk 2]
```diff
@@ -30,16 +30,62 @@ def _build_subfolder(self):
...
+        # OpenAL's API is pure C, thus the c++ standard does not matter
+        # Because the backend is C++, the C++ STL matters
         del self.settings.compiler.cppstd
+        if not self._openal_cxx_backend:
+            del self.settings.compiler.libcxx
 
-        if tools.Version(self.version) >= "1.20" and self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "5":
-            raise ConanInvalidConfiguration("OpenAL can't be compiled by {0} {1}".format(self.settings.compiler,
-                                                                                         self.settings.compiler.version))
+        if tools.Version(self.version) >= "1.22":
```
Comment (madebr) [inline recipes/openal/all/conanfile.py:77]:
```suggestion
        if tools.Version(self.version) >= "1.21":
```

File: recipes/openal/all/conanfile.py [hunk 3]
```diff
@@ -30,16 +30,65 @@ def _build_subfolder(self):
...
+    def _openal_cxx_backend(self):
+        return tools.Version(self.version) >= "1.20"
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
-        del self.settings.compiler.libcxx
+        # OpenAL's API is pure C, thus the c++ standard does not matter
+        # Because the backend is C++, the C++ STL matters
         del self.settings.compiler.cppstd
```
Comment (SpaceIm) [inline recipes/openal/all/conanfile.py:76]: But you can still compile openal with a different standard, so compiler.cppstd matters.

Comment (madebr) [reply recipes/openal/all/conanfile.py:76]: The cppstd will only change the internals. The external will not change a thing.
I don't think people cares whether openal is compiled with c++14 or c++20.

Keeping libcxx will also requires adding patches removing the hard coding of the c++ standard.
e.g. https://github.com/kcat/openal-soft/blob/d062c16629a93e37366ab88d60d2df3372a234e2/CMakeLists.txt#L153-L155
I don't think this is worth the effort.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**